### DISCUSSION TITLE
Logging of incoming and outgoing messages from other IDS infrastructure components

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-edc/Connector#2644] [Q&A] [answered]
Is it possible to log-output incoming and outgoing IDS-messages from and to other IDS infrastructure components (e.g. central Broker) in the log?

Use case: The other IDS infrastructure component (e.g. Broker) responds with a RejectionMessage instead of an expected MessageProcessedNotificationMessage and we want to log-output the entire IDS-message response (and in advance send IDS-message).

(edc-known-friend edc-broker-extension would need this: https://github.com/sovity/edc-extensions/tree/main/extensions/broker)

**Comment by jimmarino** [comment, marked as answer]
I don't think this should be part of the core repository or encouraged as a best practice:

1. The broker is not part of the IDS Dataspaces Protocol. The Catalog specifications have superseded the broker concept.
2. Sensitive information may be contained in incoming and outgoing messages.
3. Monitoring is not the correct place to intercept ingress and egress points in the runtime.

If you want to send some type of audit event, the eventing system is the correct place to do this. If you are interested in recording messages for debugging purposes, Jersey filters and OkHttp interceptors are the way to go.

**Reply by ghost** [reply]
Thank you :)
